### PR TITLE
refactor message generation

### PR DESCRIPTION
**Pull request by dirk-thomas**
Connects to ros2/ros2#48.

### CODE REVIEW

**Comment by esteve** [comment]
+1

File: userland/src/add_two_ints_client.cpp [hunk 1]
```diff
@@ -37,8 +37,10 @@ int main(int argc, char ** argv)
 
   auto node = rclcpp::Node::make_shared("add_two_ints_client");
 
-  auto client = node->create_client<userland_msgs::AddTwoInts>("add_two_ints");
-  auto request = std::make_shared<userland_msgs::AddTwoInts::Request>();
+  auto client = node->create_client<
+userland_msgs::srv::AddTwoInts>("add_two_ints");
+  auto request = std::make_shared<
+userland_msgs::srv::AddTwoInts::Request>();
```
Comment (wjwwood) [inline userland/src/add_two_ints_client.cpp]: [style] Why are these lines lacking any leading whitespace? I would expect them to the be indented once from the indention level of the line they started on. (similar cases through out the pr)

Comment (dirk-thomas) [inline userland/src/add_two_ints_client.cpp]: Fixed 3716b39facc64c29b8a7577643d60b2c668c3ee6. Same for server part.

**Comment by wjwwood** [comment]
+1

**Comment by tfoote** [comment]
+1

**Pull request merged**